bundle/direct/dresources/experiment.go

```diff
@@ -0,0 +1,91 @@
+package dresources
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/cli/bundle/deployplan"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/ml"
+)
+
+type ResourceExperiment struct {
+	client *databricks.WorkspaceClient
+}
+
+func (*ResourceExperiment) New(client *databricks.WorkspaceClient) *ResourceExperiment {
+	return &ResourceExperiment{
+		client: client,
+	}
+}
+
+func (*ResourceExperiment) PrepareState(input *resources.MlflowExperiment) *ml.CreateExperiment {
+	return &ml.CreateExperiment{
+		Name:             input.Name,
+		ArtifactLocation: input.ArtifactLocation,
+		Tags:             input.Tags,
+		ForceSendFields:  filterFields[ml.CreateExperiment](input.ForceSendFields),
+	}
+}
+
+func (*ResourceExperiment) RemapState(experiment *ml.Experiment) *ml.CreateExperiment {
+	return &ml.CreateExperiment{
+		Name:             experiment.Name,
+		ArtifactLocation: experiment.ArtifactLocation,
+		Tags:             experiment.Tags,
+		ForceSendFields:  filterFields[ml.CreateExperiment](experiment.ForceSendFields),
+	}
+}
+
+func (r *ResourceExperiment) DoRefresh(ctx context.Context, id string) (*ml.Experiment, error) {
+	result, err := r.client.Experiments.GetExperiment(ctx, ml.GetExperimentRequest{
+		ExperimentId: id,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get experiment %s: %w", id, err)
```

Comment:
The error wrapping would only be needed if there is ambiguity / multiple operations. If it 1-1 mapping with SDK method, you could just return error, we have enough context added by the framework, including resource type, key, id etc.